### PR TITLE
Node server/integration

### PR DESCRIPTION
**Pull request by aayushdhiman01**
## Pull Request Details
Integrated SignUp , SignIn of both Doctor and Patient.

### Description

Node Server storing patient and doctor signup and login data to MongoDb database. Need some help in integrating with frontend as it is not hitting the api on clicking the button. I checked with postman the apis were working fine and storing data in backend.

## Summary by CodeRabbit

- **New Features**
  - Integrated `dotenv` package for environment variable management.

- **Updates**
  - Updated `LoginForm` API endpoint to `/api/auth/login`.
  - Updated `SignupForm` API endpoint to `/api/auth/register`.

- **Chores**
  - Updated dependencies in `package-lock.json` to include new versions of `@types/bcrypt`, `@types/node`, and `undici-types`.

### CODE REVIEW

**Review by Latish705 — COMMENTED** [review]
Why so many packages

**Comment by aayushdhiman01** [comment]
@Latish705 dotenv is for storing the links of database and jwt secret key which can be done in some other way without using package. @types packages are for working with typescript.
Any changes i need to make???

File: node_server/src/app.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const app = express()
+
+app.use(express.json());
+
+app.use(cors({ origin: process.env.CLIENT_URL, credentials: true }));
+
+app.use("/api/auth",authUserRoute);
+app.use("/api/authDoctor",authDoctorRoute);
+
+app.use(bodyParser.urlencoded({extended:true}));
```
Comment (Latish705) [inline node_server/src/app.ts:31]: this body parser sometime give error we can keep it above all the routes

File: node_server/src/app.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    .then(( ()=> console.log("MongoDb Connected")))
+    .catch((err) => console.log("Mongo Err",err));
+
+const app = express()
+
+app.use(express.json());
+
+app.use(cors({ origin: process.env.CLIENT_URL, credentials: true }));
+
+app.use("/api/auth",authUserRoute);
```
Comment (Latish705) [inline node_server/src/app.ts:30]: we are going to build it to large scale so have routes like api/user/auth etc will be good

**Review by Latish705 — APPROVED** [review]
@aayushdhiman01 Really good job!👍 need some small changes

File: node_server/src/app.ts (hunk 1 above)
Comment (aayushdhiman01) [reply node_server/src/app.ts]: done moved above the routes

**Comment by Latish705** [comment]
@aayushdhiman01 can you show the video of how the login and signup is working with node

File: node_server/src/app.ts (hunk 2 above)
Comment (aayushdhiman01) [reply node_server/src/app.ts]: done made the changes to api/user/auth and api/doctor/auth

**Review by Latish705 — APPROVED** [review]
Nice job

**Comment by aayushdhiman01** [comment]
@Latish705 actually there is some issue in connection with frontend as on clicking the signup button it is not hitting the APIs basically frontend and backend are not properly connected, but APIs are working fine as i checked using postman , data passed was getting stored to the backend.

**Comment by aayushdhiman01** [comment]
postman

**Comment by Latish705** [comment]
@aayushdhiman01 Thanks for contribution!

**Pull request merged**